### PR TITLE
Adds a quick way of provisioning a validator in GCP

### PR DESCRIPTION
**Pull request by brunocalza**

### CODE REVIEW

File: infra/main.tf [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+variable "credentials_file" {
+    type = string
+}
+
+provider "google" {
+    credentials = file("${var.credentials_file}")
+
+    project = "textile-310716"
+    region  = "us-west1"
+    zone    = "us-west1-b"
```
Comment (jsign) [inline infra/main.tf:27]: Maybe makes sense to make these also `${var.***}`?

File: infra/main.tf [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+resource "google_compute_instance" "validator" {
+    name         = "${var.vm_name}"
+    machine_type = "e2-standard-2"
+    tags         = ["https-server", "grafana", "ssh-vm"]
+    deletion_protection = false
```
Comment (jsign) [inline infra/main.tf:34]: In our infra docs we suggest setting this to `true`. 
No strong opinion in doing the same here, just checking if was intentional.

File: infra/main.tf [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            image = "ubuntu-2204-jammy-v20220902"
+            type = "pd-ssd"
+            size = "30"
+        }
+    }
+    
+    network_interface {
+        network = "default"
+        access_config {}
+    }
```
Comment (jsign) [inline infra/main.tf:47]: Prob out of scope for Terraform, but we should remember (maybe comment here?) that the operator should set up firewall rules, but perhaps that's somewhat obvious.

**Review by jsign — APPROVED** [review]
Great! :100:

File: infra/main.tf (hunk 3 above)
Comment (brunocalza) [reply infra/main.tf:47]: We can set the firewall rules here or add a `depends on` so Terraform checks if firewall rule exists

**Review by jsign — DISMISSED** [review]

**Review by jsign — APPROVED** [review]

**Pull request merged**